PoSSapi/src/Application/Businesses/Commands/CreateBusinessCommand.cs

```diff
@@ -0,0 +1,49 @@
+using AutoMapper;
+using MediatR;
+using PoSSapi.Application.Common.Exceptions;
+using PoSSapi.Application.Common.Interfaces;
+using PoSSapi.Domain.Entities;
+
+namespace PoSSapi.Application.Businesses.Commands;
+
+public class CreateBusinessCommand : IRequest<Guid>
+{
+    public Guid Id { get; init; }
+    public Guid? ManagerId { get; init; }
+}
+
+public class CreateBusinessCommandHandler : IRequestHandler<CreateBusinessCommand, Guid>
+{
+    private readonly IApplicationDbContext _context;
+
+    public CreateBusinessCommandHandler(IApplicationDbContext context, IMapper mapper)
+    {
+        _context = context;
+    }
+
+    public async Task<Guid> Handle(CreateBusinessCommand request, CancellationToken cancellationToken)
+    {
+        User? manager = null;
+        
+        if (request.ManagerId.HasValue)
+        {
+            manager = await _context.Users.FindAsync(request.ManagerId.Value);
+            if (manager == null)
+            {
+                throw new NotFoundException(nameof(User), request.ManagerId.Value);
+            }
+        }
```

Comment:
Having ManagerId while creating a business seems like an impossible scenario, because an user for that business would have to exist (but you can't create an user for a business that has not yet been created), unless you take an user of another business, but that seems conflicting. Perhaps ManagerId shouldn't be an available parameter in the create business command?

Reply:
True, this should probably only be available in updating the business